pkg/ansible/controller/reconcile.go

```diff
@@ -306,16 +318,12 @@ func (r *AnsibleOperatorReconciler) markError(u *unstructured.Unstructured, name
 }
 
 func (r *AnsibleOperatorReconciler) markDone(u *unstructured.Unstructured, namespacedName types.NamespacedName, statusEvent eventapi.StatusJobEvent, failureMessages eventapi.FailureMessages) error {
-	logger := logf.Log.WithName("markDone")
 	// Get the latest resource to prevent updating a stale status
 	err := r.Client.Get(context.TODO(), namespacedName, u)
-	if apierrors.IsNotFound(err) {
-		logger.Info("Resource not found, assuming it was deleted", err)
-		return nil
-	}
```

Comment:
This change would result in returning an error from this function when the CR is not found, which would result in returning an error from the main `Reconcile` function (see [here](https://github.com/operator-framework/operator-sdk/blob/bb3ad277dea92396f10f573b1443999fcada088c/pkg/ansible/controller/reconcile.go#L234-L240)).

By returning the not found error, we would trigger another Reconcile unnecessarily. By returning `nil`, the `Reconcile` function will also return `nil`, which will result in no further reconciliations occurring.

Ditto for the changes in `markError`, except that we never actually check the error returned from `markError`. In the `markError` cases, we always return the original error no matter what happens with `markError`.